version: '3.8'

services:
  django_gunicorn:
    build: ./devops/gunicorn
    volumes:
    - static:/static
    - .:/app

    ports:
    - "8000:8000"

    depends_on:
      db:
        condition: service_healthy

    environment:
    - POSTGRES_DB=postgres
    - POSTGRES_USER=postgres
    - POSTGRES_PASSWORD=postgres
    - SECRET_KEY='c*48+8)np*$4ohvxr38x=)(w4vw3!shqm*+m+%v$uzn!q@wk%+'
    - DEBUG=False
    - DJANGO_ENVIRONMENT=production
  nginx:
    build: ./devops/nginx
    volumes:
    - static:/static
    ports:
    - "80:80"
    depends_on:
    - django_gunicorn
  db:
    image: postgres
    environment:
    - POSTGRES_DB=postgres
    - POSTGRES_USER=postgres
    - POSTGRES_PASSWORD=postgres
    volumes:
    - data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d $${POSTGRES_DB} -U $${POSTGRES_USER}" ]
      interval: 10s
      timeout: 5s
      retries: 5
volumes:
  static:
  data:
